sprites/tagalong.asm

```diff
@@ -0,0 +1,1614 @@
+    dw $0178, $0550, $0001, $0023, $0001
+    dw $0168, $04F8, $0002, $002A, $0001
+    dw $1BD8, $16FC, $0001, $0124, $0006
+    dw $1520, $167C, $0001, $0124, $0006
+    dw $05AC, $04FC, $0001, $0029, $0001
+    
+    ; $4A54E
+    .overworld_id
+    dw $0003 ; OW 03 
+    dw $005E ; OW 5E
+    dw $0000 ; OW 00
+    
+    ; $04A554 to $4A585
+    .area_data_1
+    dw $03C0, $0730, $0001, $009D, $0004
+    dw $0648, $0F50, $0000, $FFFF, $000A
+    dw $06C8, $0D78, $0001, $FFFF, $000A
+    dw $0688, $0C78, $0002, $FFFF, $000A
+    dw $00E8, $0090, $0000, $0028, $000E
+    
+    ; $04A586 ($4A588 too, in a way)
+    .room_data_boundaries_1
+    dw 0, 30, 60, 70, 90, 100, 110, 120,
+    
+    ; $4A596
+    .area_data_boundaries_1
+    dw 0, 10, 40, 50
+}
+
+; ==============================================================================
+
+; $04A59E-$04A6CC LOCAL JUMP LOCATION
+Tagalong_HandleTrigger:
+{
+    LDA $11 : BNE .no_text_message
+      REP #$30
+      LDY #$0000
+      LDA $1B : AND.w #$00FF : BEQ .check_overworld_triggers
+        INY
+        LDX.w #$000C
+        LDA $A0
+        .check_next_room
+        CMP Tagalong_TriggerData_room_id, X : BEQ .room_match
+          DEX #2 : BPL .check_next_room
+        BRA .no_text_message
+      .check_overworld_triggers
+      
+      LDX.w #$0004
+      LDA.b $8A
+      
+      .check_next_area
+      
+      ; Select graphics based on certain areas maybe?
+      ; the areas mentioned in this array are the mountain, the forest,
+      ; and the maze in the dark world (i.e. ???, old man, and kiki)
+      CMP Tagalong_TriggerData_overworld_id, X : BEQ .area_match
+      
+      DEX #2 : BPL .check_next_area
+    
+    .no_text_message
+    
+    BRL .return
+    
+    .room_match
+    
+    LDA.w Tagalong_TriggerData_room_data_boundaries_1+2, X : STA $08
+    
+    LDA.w Tagalong_TriggerData_room_data_boundaries_1, X : TAX
+    
+    .next_room_data_block
+    
+    STX $0C
+    
+    STZ $0A
+    
+    LDA $7EF3CC : AND.w #$00FF : CMP $A4DE, X : BNE .not_room_data_match
+      LDA.w Tagalong_TriggerData_dungeon_coordinates, X : STA $00
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+2, X : STA $02
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+4, X : STA $06
+      LDA.w Tagalong_TriggerData_dungeon_coordinates+8, X : STA $04
+      
+      SEP #$30
+      
+      JSR Tagalong_CheckTextTriggerProximity : BCS .check_flags_and_proximity
+      
+      REP #$30
+    
+    .not_room_data_match
+    
+    LDA $0C : CLC : ADC.w #$000A : TAX
+    
+    CPX $08 : BNE .next_room_data_block
+    
+    BRL .return
+    
+    .area_match
+    
+    LDA.w Tagalong_TriggerData_area_data_boundaries_1+2, X : STA $08
+    LDA.w Tagalong_TriggerData_area_data_boundaries_1, X : TAX
+    
+    .next_area_data_block
+    
+    STX $0C
+    STZ $0A
+    
+    LDA $7EF3CC : AND.w #$00FF : CMP $A55C, X : BNE .not_area_data_match
+    
+    LDA.w Tagalong_TriggerData_area_data_1, X : STA $00
```

Comment:
these are wrong i'm pretty sure.
 it should be +0, +2, +4, and +6 not +0, +2, +6, +8
 
 ```
LDA.w $A554, X : STA $00
LDA.w $A556, X : STA $02
LDA.w $A558, X : STA $06
LDA.w $A55A, X : STA $04
```

Reply:
Done.